api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -701,11 +714,29 @@ private boolean userNeverLoggedIn(com.google.api.services.admin.directory.model.
     }
     JWTWrapper wrapper = new JWTWrapper().jwt(token.getJwt());
     try {
-      fireCloudService.postNihCallback(wrapper);
+      NihStatus nihStatus = fireCloudService.postNihCallback(wrapper);
+      if (nihStatus.getLinkedNihUsername() == null) {
+        log.log(Level.WARNING, "No Username Found when updating nih token");
+      }
       User user = initializeUserIfNeeded();
+      if (nihStatus != null) {
+        Timestamp eraCommonsCompletionTime = user.getEraCommonsCompletionTime();
+        if ((nihStatus.getLinkedNihUsername() != null &&
+            !nihStatus.getLinkedNihUsername().equals(user.getEraCommonsLinkedNihUsername())) ||
+            nihStatus.getLinkExpireTime() != user.getEraCommonsLinkExpireTime().getTime()) {
+          eraCommonsCompletionTime = new Timestamp(clock.instant().toEpochMilli());
+        }
+        user = userService.setEraCommonsStatus(
+            nihStatus.getLinkedNihUsername(),
+            new Timestamp(nihStatus.getLinkExpireTime()),
+            eraCommonsCompletionTime
+        );
+      } else {
+        user = userService.setEraCommonsStatus(null, null, null);
```

Comment:
Do we need this or will these values be explicitly null we initialize the user here ?

Reply:
Initialize user if needed won't always create a new user. If there is already a user it will do nothing, so we still need this set.